### PR TITLE
Add API to set the IEEE 802.15.4 Extended Address.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/openthread.cpp [hunk 1]
```diff
@@ -96,6 +96,19 @@ const uint8_t *otGetExtendedAddress(void)
     return reinterpret_cast<const uint8_t *>(sThreadNetif->GetMac().GetExtAddress());
 }
 
+ThreadError otSetExtendedAddress(const otExtAddress *aExtAddress)
+{
+    ThreadError error = kThreadError_None;
```
Comment (librasungirl) [inline src/core/openthread.cpp]: Would we consider to verify aExtAddress to ensure u/l is 1 and i/g is 0 before making it effect?

Comment (jwhui) [inline src/core/openthread.cpp]: I think it makes sense to enforce the i/g bit.  I'm not sure we want to enforce the u/l bit yet.  The client may want to clear the u/l bit for some reason.

Comment (jwhui) [inline src/core/openthread.cpp]: Just submitted an update that checks the i/g bit.

**Comment by mszczodrak** [comment]
LGTM :+1:

**Pull request merged**